pinc/ProofProject.inc

```diff
@@ -0,0 +1,217 @@
+<?php
+declare(strict_types=1);
+
+class ProofProjectPage extends LPage
+{
+    public function __construct(object $proof_project, object $project_page)
+    {
+        global $pguser;
+
+        $this->page_state = $project_page->page_state;
+        $this->page_name = $project_page->page_name;
+        $this->project = $proof_project->project;
+        $this->projectid = $proof_project->project->projectid;
+        $this->round = $proof_project->round;
```

Comment:
There is also a round variable inside Lpage which is derived from the page_state. Although they will be the same this is not good. Perhaps we should consider revising LPage so that the constructor receives a round (which is easily avaiable in the old code derived from project_state)